### PR TITLE
class hierarchy document

### LINKED ISSUE #571: Outline Dev Guide Section - Class Hierarchy (Tables)
See `R/00tabletrees.R`

### PR DESCRIPTION
**Pull request by ayogasekaram**
Initial document with subsection list.

closes #571

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -47,6 +47,15 @@ articles:
...
 reference:
+<<<<<<< HEAD
+
+  - title: For Developers
+    desc: Vignettes aimed at package developers
+    contents:
+    - dg_table_hierarchy
+    
+=======
+>>>>>>> main
```
Comment (Melkiades) [inline _pkgdown.yml:58]: to be fixed

File: inst/dev-guide/dg_table_hierarchy.Rmd [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+## Tree Paths 
+
+Root to Leaves, are vectors of vectors 
+Tables are tree, nodes in the tree can have summaries associated with them. Tables are trees because of the nested structure. There is also the benefit of keeping and repeating necessary information when trying to paginate a table.
+
+Children of ElementaryTables are row objects. TableTree can have children that are either row objects or other table objects.
+
+
+#### TODO:
+Create Tree Diagram showing class hierarchy.
```
Comment (Melkiades) [inline inst/dev-guide/dg_table_hierarchy.Rmd:98]: any news regarding this?

File: vignettes/dev-guide/dg_table_hierarchy.Rmd [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+## Tree Paths 
+
+Root to Leaves, are vectors of vectors 
+Tables are tree, nodes in the tree can have summaries associated with them. Tables are trees because of the nested structure. There is also the benefit of keeping and repeating necessary information when trying to paginate a table.
+
+Children of ElementaryTables are row objects. TableTree can have children that are either row objects or other table objects.
+
+
+#### TODO:
+Create Tree Diagram showing class hierarchy.
```
Comment (Melkiades) [inline vignettes/dev-guide/dg_table_hierarchy.Rmd:95]: @ayogasekaram Is this the only missing thing? I think we can eventually reiterate updates in the future, what do you think? ^^

**Review by Melkiades — APPROVED** [review]
Lgtm! Thanks Abi ;) when you think it is ok we can merge it. I think it is better to reiterate updates

File: vignettes/dev-guide/dg_table_hierarchy.Rmd (hunk 3 above)
Comment (edelarua) [reply vignettes/dev-guide/dg_table_hierarchy.Rmd:95]: @ayogasekaram will you be adding this section in a later PR? If so, I think it's good to go for now.

**Pull request merged**